python/pyarrow/tests/test_types.py

```diff
@@ -1331,7 +1331,7 @@ def __init__(self, schema):
         def __arrow_c_schema__(self):
             return self.schema.__arrow_c_schema__()
 
-    schema = pa.schema([pa.field("field_name", pa.int32())])
+    schema = pa.schema([pa.field("field_name", pa.int32())], metadata={"a", "b"})
     wrapped_schema = Wrapper(schema)
 
     assert pa.schema(wrapped_schema) == schema
```

Comment:
What you additionally need to test is `pa.schema(wrapped_schema, metadata=..)`, i.e. passing metadata to the `pa.schema(..)` constructor function when passing a schema object to it (the test as you edited it now will already pass, I think, and so not cover the regression)

Reply:
Ah, thanks. I tried pushing just the test first hoping CI would run them but got lazy when it didn't. :)